cpp/find_latest_journal_year.cc

```diff
@@ -157,12 +157,28 @@ void ProcessRecords(MARC::Reader * const marc_reader, std::unordered_map<std::st
             continue;
 
         std::string year_as_string;
-        for (const auto &field : record.getTagRange("936")) {
-            const MARC::Subfields subfields(field.getSubfields());
-            year_as_string = subfields.getFirstSubfieldWithCode('j');
-            if (not year_as_string.empty())
-                break;
+
+        std::vector<std::string> subfields;
+        std::vector<std::string> filtered_dates;
+        for (const auto &field : record.getTagRange("773")) {
+            if (field.getIndicator1() == '1') {
+                for (const auto &subfield : field.getSubfields()) {
+                    StringUtil::Split(subfield.value_, ':', &subfields, true);
+                    filtered_dates.emplace_back(subfields[1]);
+                }
+            }
         }
+        if (not filtered_dates.empty()) {
+            year_as_string = filtered_dates[1];
```

Comment:
I don't really understand the logic at this point.
Usually filtered_dates contains only 1 value, but then you must use filtered_dates[0].
On the other hand, the name of this program is "find_latest_journal_year". That means that if filtered_dates contain multiple values, you must detect the biggest number out of all values in the container.
Another option is: I doubt that there will be multiple 773 fields with the same indicators, so you could as well remove the whole filtered_dates variable and write the resullt to year_as_string instead and then just use the existing logic for that variable.